numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2100,6 +2100,7 @@ array_fromstring(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds
     array_function_result = array_implement_c_array_function_creation(
             "fromstring", args, keywds);
     if (array_function_result != Py_NotImplemented) {
+        Py_XDECREF(descr);
```

Comment:
I had already fixed that in the larger argument parsing, so this was also not part of the initial PR.